meson.build

```diff
@@ -0,0 +1,3419 @@
+    conf.set('HAVE_' + decl.underscorify().to_upper(), 1)
+  endif
+endforeach
+
+decl_headers = '''
+#include <inttypes.h>
+#include <netinet/in.h>
+#include <netinet/ether.h>
+#include <linux/rtnetlink.h>
+#include <net/if.h>
+#include <linux/ip.h>
+#include <linux/if_tunnel.h>
+#include <linux/if_link.h>
+#include <linux/if_bridge.h>
+#include <linux/if_addr.h>
+#include <linux/neighbour.h>
+'''
+
+foreach decl : ['IFLA_INET6_ADDR_GEN_MODE',
+                'IN6_ADDR_GEN_MODE_STABLE_PRIVACY',
+                'IFLA_VRF_TABLE',
+                'IFLA_MACVLAN_FLAGS',
+                'IFLA_IPVLAN_MODE',
+                'IFLA_PHYS_PORT_ID',
+                'IFLA_BOND_AD_INFO',
+                'IFLA_VLAN_PROTOCOL',
+                'IFLA_VXLAN_REMCSUM_NOPARTIAL',
+                'IFLA_VXLAN_GPE',
+                'IFLA_VTI_REMOTE',
+                'IFLA_IPTUN_ENCAP_DPORT',
+                'IFLA_GRE_ENCAP_DPORT',
+                'IFLA_BRIDGE_VLAN_INFO',
+                'IFLA_BRPORT_PROXYARP',
+                'IFLA_BRPORT_LEARNING_SYNC',
+                'IFLA_BR_VLAN_DEFAULT_PVID',
+                'NDA_IFINDEX',
+                'IFA_FLAGS',
+               ]
+  conf.set10('HAVE_DECL_' + decl,
+             compiler.compiles(decl_headers + 'int x = ' + decl + ';'))
+endforeach
+
+skip = false
+foreach ident : ['secure_getenv', '__secure_getenv']
+  if not skip and compiler.has_function(ident)
+    conf.set('HAVE_' + ident.to_upper(), 1)
+    skip = true
+  endif
+endforeach
+
+decl_test_format = '''
+#define _GNU_SOURCE
+#include <fcntl.h>
+#include <linux/loop.h>
+#include <linux/random.h>
+#include <sched.h>
+#include <string.h>
+#include <sys/mount.h>
+#include <sys/types.h>
+#include <unistd.h>
+int main(int argc, char **argv) {
+    void *x = &@0@;
+    return 0;
+}
+'''
+
+foreach ident : [
+  'memfd_create',
+  'gettid',
+  'pivot_root',
+  'name_to_handle_at',
+  'setns',
+  'getrandom',
+  'renameat2',
+  'kcmp',
+  'keyctl',
+  'copy_file_range',
+  'explicit_bzero',
+  'LO_FLAGS_PARTSCAN',
+  ]
+  have = compiler.compiles(decl_test_format.format(ident))
+  message('@0@ is declared: @1@'.format(ident, have))
+  conf.set10('HAVE_DECL_' + ident.to_upper(), have)
+endforeach
+
+#####################################################################
+
+sed = find_program('sed')
+grep = find_program('grep')
+awk = find_program('awk')
+m4 = find_program('/usr/bin/m4')
+stat = find_program('stat')
+
+progs = [['quotaon',    '/usr/sbin/quotaon'   ],
+         ['quotacheck', '/usr/sbin/quotacheck'],
+         ['kill',       '/usr/bin/kill'       ],
+         ['kmod',       '/usr/bin/kmod'       ],
+         ['kexec',      '/usr/sbin/kexec'     ],
+         ['sulogin',    '/usr/sbin/sulogin'   ],
+         ['mount',      '/usr/bin/mount',     'MOUNT_PATH'],
+         ['umount',     '/usr/bin/umount',    'UMOUNT_PATH'],
+        ]
+
+foreach prog : progs
+  exe = find_program(prog[0], required: false)
```

Comment:
@jpakkane the binaries that are checked for here are not strictly required for building systemd. It's just that we need the path to the binaries as those are encoded in a few places. Under configure.ac we have checks like
```
AC_PATH_PROG([KEXEC], [kexec], [/usr/sbin/kexec], [$PATH:/usr/sbin:/sbin])
```

When building the Debian package, we do not actually install the kexec-tools package to get the path, but instead we set it explicitly like this

```QUOTAON=/sbin/quotaon QUOTACHECK=/sbin/quotacheck SETCAP=/sbin/setcap .. ./configure ...```

Can we do something similar with meson and ```find_program```?